hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java

```diff
@@ -33,6 +34,7 @@
  * NodeExtension is a <tt>Node</tt> extension mechanism to be able to plug different implementations of
  * some modules, like; <tt>SerializationService</tt>, <tt>SocketChannelWrapperFactory</tt> etc.
  */
+@PrivateApi
 public interface NodeExtension {
```

Comment:
NodeExtension can't be public API because it is treated as private. We do not care about compatibility etc.